internal/config/config_test.go

```diff
@@ -26,3 +26,115 @@ func TestConfig_Validate_WhenFilterInvalid_ReturnsError(t *testing.T) {
 	// Assert
 	g.Expect(err).To(HaveOccurred())
 }
+
+func TestConfig_Validate_WhenModeIsEmpty_SetsDefaultMode(t *testing.T) {
+	t.Parallel()
+	g := NewGomegaWithT(t)
+
+	// Arrange
+	cfg := &Config{
+		Mode: "",
+	}
+
+	// Act
+	err := cfg.Validate()
+
+	// Assert
+	g.Expect(err).ToNot(HaveOccurred())
+	g.Expect(cfg.Mode).To(Equal("single-file"))
+}
+
+func TestConfig_Validate_WhenModeIsValidSingleFile_NormalizesMode(t *testing.T) {
+	t.Parallel()
+
+	testCases := []string{
+		"single-file",
+		"Single-File",
+		"SINGLE-FILE",
+		"singlefile",
+		"SingleFile",
+		"SINGLEFILE",
```

Comment:
Don't need these test cases - checks should be case insensitive, but it's ok to return an error if extra characters are present - or required characters are missing.

Reduce the list of test cases to `single-file`, `Single-File` and `SINGLE-FILE`.